synop2bufr/__init__.py

```diff
@@ -1217,7 +1218,18 @@ def extract_individual_synop(data: str) -> list:
             "Invalid SYNOP message: AAXX could not be found."
         )
 
-    data = re.split('(AAXX [0-9]{5})', data[start_position:])
+    # Split the string by AAXX YYGGiw
+    data = re.split(r'(AAXX\s+[0-9]{5})', data[start_position:])
+
+    # Check if the beginning of the message, that we're about to throw
+    # away (data[0]), also contains AAXX and thus there must be a
+    # typo present at the AAXX YYGGiw part of the report
+    if data[0].__contains__("AAXX"):
```

Comment:
"AAXX" in data[0]?

Reply:
Since the message is split around AAXX YYGGiw (with an indefinite number of spaces between AAXX and YYGGiw as sometimes there is an extra space by accident), data[0] would be the part of the message before this section (e.g. ZCZC 123 SMCU20 MUHV 310000). This is normally discarded, but if it contains 'AAXX', then that means the splitting of the message has gone wrong due to a typo somewhere. There was a bad piece of data that motivated this change but it's been so long now I can't remember.

Reply:
I understand that, but why are we using `__contains__` rather than `in`?

Reply:
They are equivalent but I will change it to `in`, I can't remember why I put `__contains__`.